### ISSUE TITLE
No collectTotals() called on quote when returning from wirecards server

### ISSUE THREAD

**Issue by archonkulis** [wirecard/magento2-wcs#14]
In our client's project, there were some calculations with orders items original_price.

Everything was good with other payment methods, except when paid with this extensions payment method, because original_price in sales_order_item was null.

By comparing a lot of data with other payment methods, where original_price got filled, I noticed, that the difference was that this extension isn't calling collectTotals() on quote object at any time.

The file I changed as a quick fix is following:

```
vendor/wirecard/magento2-wcs/Controller/Checkout/Start.php
function execute()
```

straight after `$payment = $this->_cart->getQuote()->getPayment()->getMethodInstance(); `

I wrote:

`$this->_cart->getQuote()->collectTotals();`

( of course by extending )

Now it seems to add original_price and any other price related stuff.

Could you please check if this is the correct place to add the totals collection ( since you know extensions inner workings better ) and let me know why you didn't include it in the first place - just wondering, maybe there was a special reason for that and I should avoid using it?

Will wait for the feedback.

Thanks and have a nice day!

**Comment by jakubpolomsky** [comment]
Hello!

Thank you for your feedback. To answer your question in a logical order I'll start with the place.

1. Actually any place before saving the quote is correct. The class Start is a pretty good candidate.
2. Why aren't we using this method? We do not need it and you are the first who'd like to have it included. There is no reason to avoid it. We'll implement it in the next release alongside other updates after thorough testing.
3. In the meantime, please feel free to modify the code and if you think your changes are profitable for the community, please create a pull request with your changes.

Thank you once again for your feedback!

**Comment by archonkulis** [comment]
Just FYI:

When I added collectTotals in class Start, it didn't work when the order was created AFTER payment. It did work, if order was created before, though.

So, I included in that file to collectTotals only if:

`            if ($this->_dataHelper->getConfigData('options/order_creation') == 'before') {
                $this->_cart->getQuote()->collectTotals();
            }`

And in file vendor/wirecard/magento2-wcs/Model/OrderManagement.php after this line:

`} else if ($return->mage_orderCreation == 'after') {`

I included following:

`$quote->collectTotals();`

( by extending of course )

Now it seems to work with both settings.
